### PR TITLE
fix(sample-gen): should return xml literal example

### LINKED ISSUE #6818: A string example for an XML schema crashes the example renderer
### Q&A (please complete the following information)
 - OS: Windows 10
 - Browser: Chrome, Firefox latest
 - Method of installation: dist assets
 - Swagger-UI version: 3.39.0
 - Swagger/OpenAPI version: OpenAPI 3.0 (I didn't check OAS2 but it may also be affected)

### Content & configuration
Example Swagger/OpenAPI definition:
```yaml
openapi: 3.0.3
info:
  title: XML examples
  version: 1.0.0
paths:
  /foo:
    get:
      responses:
        '200':
          description: ok
          content:
            application/xml:
              schema:
                $ref: '#/components/schemas/MyObject'
              examples:
                asString:
                  value: |-
                    <MyObject id="5">
                      <foo>bar<foo>
                    </MyObject>
        '201':
          description: ok
          content:
            application/xml:
              schema:
                $ref: '#/components/schemas/MyObject'
              example: |-
                <MyObject id="5">
                  <foo>bar<foo>
                </MyObject>
        '202':
          description: ok
          content:
            application/xml:
              schema:
                $ref: '#/components/schemas/MyObject'

components:
  schemas:
    MyObject:
      type: object
      properties:
        id:
          type: string
          xml:
            attribute: true
        foo:
          type: string
      example: |-
        <MyObject id="5">
          <foo>bar<foo>
        </MyObject>
```

### Describe the bug you're encountering
An XML object schema with a text/string example crashes the example renderer.
v. 3.35.2 did not have this issue.

### Expected behavior
When an XML schema has a text/string example, that example should be displayed as is.

### Screenshots
image

### Additional context or thoughts
The example renderer was reworked in v. 3.36.0 and later releases. This issue was most likely introduced by those changes.

### PR DESCRIPTION
**Pull request by mathis-m**
### Description

A string example for an XML schema should be returned as provided.
In `sampleFromSchemaGeneric` added in case xml object schema the conditional return for the string example.
Added in `createXMLExample` the conditional return for the string example.

### Motivation and Context

Fixes #6818

### How Has This Been Tested?

added unit test to ensure literal example is returned as is.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/core/plugins/samples/fn.js [hunk 1]
```diff
@@ -233,6 +233,10 @@ export const sampleFromSchemaGeneric = (schema, config={}, exampleOverride = und
 
     // generate xml sample recursively for object case
     if(type === "object") {
+      // case literal example
+      if(typeof sample === "string") {
+        return sample
+      }
```
Comment (hkosova) [inline src/core/plugins/samples/fn.js:239]: Hi @mathis-m, thanks for looking into this!

I think this check for literal examples should also be added to `type: array` schemas - i.e. for any non-primitive schemas. What do you think?

Use case:
```yaml
type: array
xml:
  name: items
  wrapped: true
items:
  type: string
  xml:
    name: item
example: |-
  <items>
    <foo>foo<foo>
    <foo>bar<foo>
  </items>
```

Comment (mathis-m) [reply src/core/plugins/samples/fn.js:242]: done

**Comment by tim-lai** [comment]
@mathis-m can you fix the merge conflict in this PR? Looks like I am merging out of sequence. =)

**Comment by mathis-m** [comment]
@tim-lai I accidently added tests on exact same lines and they don't differ two much in two PRs, that is why merge is struggling.

**Comment by tim-lai** [comment]
no problem, thanks for the update!

**Pull request merged**